WordPress/gutenberg
# Navigation block: Add current-menu-item class for post type archive

##  What?

When a post type archive is added as a link in the navigation block, and it is the current page, the `current-menu-item` CSS class is missing from the `<li>` element.
It means that the item can not be styled like other current menu items.

This PR adds a condition that checks if the page is a post type archive, and if the menu item link matches the archive link.
It sets the variable `$is_active` to true. The variable is then used to add the CSS class.

Closes https://github.com/WordPress/gutenberg/issues/46090

## Testing Instructions
Enable a custom post type which has archives enabled. (I used [Yoast test helper](https://wordpress.org/plugins/yoast-test-helper/)).
In a site wide template part, create or edit a navigation block and add two links to the archive:
One as a custom link, the other as a link in a submenu.

Go to the front of the website and view the archive page for the post type.
Locate the links to the archive page in the navigation block.
Confirm that the list elements have the `current-menu-item` CSS class.
